### PR TITLE
feat: Added component ModalWindow

### PR DESCRIPTION
**Pull request by sever7788**
Possibilities: 
* Add your own 144 * 144 pixels icon;
* Display text with title;
* Display your text on buttons;
* Add your own event handlers for the two main buttons, as well as for the window close event;
* Change the colors of buttons and icons.

Functional:
* The window display can be controlled via the isDisplayed (true/false) parameter;
* The modal window can be closed in 5 different ways:
  * by pressing one of the 3 buttons;
  * By clicking on the area outside the border of the modal window or on the "ESC" key.

### CODE REVIEW

File: src/components/ModalWindow/ModalWindow.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  style,
+  className = "",
+  fontClass = styles.subtitle3,
+  icon = Icons.errorOutline,
+  title,
+  text,
+  acceptText,
+  cancelText,
+  acceptColor = colors.red200,
+  cancelColor = "transparent",
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/ModalWindow.tsx:30]: Add it to colors.

File: src/components/ModalWindow/ModalWindow.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    document.addEventListener("keydown", escFunction, false);
+    return () => {
+      document.removeEventListener("keydown", escFunction, false);
+    };
+  }, []);
+  const escFunction = (event) => {
+    if (event.keyCode === 27) {
+      onClose();
+    }
+  };
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/ModalWindow.tsx:47]: You need describe the function before you use it.

File: src/components/ModalWindow/ModalWindow.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  const acceptOnClickHandler = (event) => {
+    acceptOnClick && acceptOnClick(event);
+    onClose();
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/ModalWindow.tsx:50]: Why you do not checked for availability of onClose() like acceptOnClick?

onClose && onClose(event);

File: src/components/ModalWindow/ModalWindow.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  };
+
+  const cancelOnClickHandler = (event) => {
+    cancelOnClick && cancelOnClick(event);
+    onClose();
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/ModalWindow.tsx:55]: Same like 50 line.

File: src/components/ModalWindow/ModalWindow.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    [styles.invisible]: !isDisplayed,
+  });
+
+  return (
+    <div style={style} className={stateStyle}>
+      <div className={styles.blur} onClick={() => onClose()} />
+      <div className={styles.modalWindow}>
+        <MetCircleIconBtn
+          onClick={() => {
+            onClose();
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/ModalWindow.tsx:69]: Add handler where you check onClose(); and 
call a function with event.

File: src/components/ModalWindow/ModalWindow.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  };
+
+  const stateStyle = classNames(styles.container, {
+    [className]: className,
+    [styles.invisible]: !isDisplayed,
+  });
+
+  return (
+    <div style={style} className={stateStyle}>
+      <div className={styles.blur} onClick={() => onClose()} />
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/ModalWindow.tsx:65]: Add handler where you check onClose(); and 
call a function with event.

File: src/components/ModalWindow/index.module.scss [hunk 7]
```diff
@@ -0,0 +1,84 @@
+@import "../styles/foundation.scss";
+
+.container {
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  display: flex;
+
+  justify-content: center;
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/index.module.scss:10]: Why you use space line?

File: src/components/ModalWindow/index.module.scss [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    transition: all 0.25s ease-in-out;
+    -webkit-backdrop-filter: blur(2px);
+    backdrop-filter: blur(2px);
+    background-color: rgba(8, 5, 5, 0.188);
+  }
+
+  .modalWindow {
+    width: 448px;
+
+    top: 50%;
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/index.module.scss:38]: Why you use space line only once?

File: src/components/ModalWindow/index.module.scss [hunk 9]
```diff
@@ -0,0 +1,84 @@
+@import "../styles/foundation.scss";
```
Comment (KarachunVladislav01) [inline src/components/ModalWindow/index.module.scss:1]: Please use space lines in styles.

File: src/components/styles/constants.scss [hunk 10]
```diff
@@ -10,3 +10,6 @@ $tranzition200: 200ms ease-in-out;
 
 // DatePicker
 $dateHeight: 32px;
+
+//Shadow
+$BigShadow: 0px 1px 2px rgba(0, 0, 0, 0.1), 0px 1px 2px rgba(0, 0, 0, 0.1), 0px 1px 8px rgba(0, 0, 0, 0.15);
```
Comment (KarachunVladislav01) [inline src/components/styles/constants.scss:15]: Haven't extra line git error.

File: src/components/styles/fonts.scss [hunk 11]
```diff
@@ -52,3 +52,10 @@
   line-height: 16px;
   font-weight: 400;
 }
+
+.H6 {
+  font-family: "Roboto", sans-serif;
+  font-size: 20px;
+  line-height: 28px;
+  font-weight: 500;
+}
```
Comment (KarachunVladislav01) [inline src/components/styles/fonts.scss:61]: Haven't extra line git error.

**Review by KarachunVladislav01 — CHANGES_REQUESTED** [review]

File: src/components/ModalWindow/ModalWindow.interface.ts [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+   */
+  className?: string;
+  /**
+   * Set title for component
+   */
+  title?: string;
+  /**
+   * Set text for component
+   */
+  text?: string;
```
[the remark this answers is not in the document]
Comment (sever7788) [reply src/components/ModalWindow/ModalWindow.interface.ts:17]: Я так и назвал "message", но Даниель сказал лучше назвать "text")

File: src/components/ModalWindow/ModalWindow.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  parameters: {
+    docs: {
+      page: mdx,
+    },
+  },
+} as Meta;
+
+
+const message = "Нажав кнопку “продолжить” вы соглашаетесь с тем, что веденные данные не будут сохранены.";
+
```
[the remark this answers is not in the document]
Comment (sever7788) [reply src/components/ModalWindow/ModalWindow.stories.tsx:24]: Ты в TabList также делал )

**Review by KarachunVladislav01 — APPROVED** [review]

**Comment by KarachunVladislav01** [comment]
Good job!

**Pull request merged**